src/apps/eth/eth_verify.c

```diff
@@ -83,15 +84,23 @@ static app_eth_sign_error_t _verify_total_fee(
 // 2) value is 0.
 app_eth_sign_error_t app_eth_verify_erc20_transaction(const ETHSignRequest* request)
 {
-    const app_eth_coin_params_t* params = app_eth_params_get(request->coin);
+    const app_eth_coin_params_t* const params = app_eth_params_get(request->coin);
     if (params == NULL) {
         return APP_ETH_SIGN_ERR_INVALID_INPUT;
     }
     const app_eth_erc20_params_t* erc20_params =
         app_eth_erc20_params_get(request->coin, request->recipient);
     if (erc20_params == NULL) {
         // unsupported token.
-        return APP_ETH_SIGN_ERR_INVALID_INPUT;
+        if (!workflow_confirm(
+                "Continue?",
+                "Unknown token\nplease verify amount\nseparately.",
+                NULL,
+                false,
+                true)) {
+            return APP_ETH_SIGN_ERR_INVALID_INPUT;
+        }
+        erc20_params = app_eth_erc20_unknown_params_get();
```

Comment:
imho this feels like a hack; the two follow up confirmation screens will probably look weird or confusing, and it is also a bit hard to read as one has to think through what empty/default params imply.

for better readability i would be more explicit:
```if (erc20_params == NULL) { return _verify_unknown_erc20(...); } ``` and not use a "fake" eth_params value.